File: 436.leetcode.py
"""
436. Find Right Interval
https://leetcode.com/problems/find-right-interval/
https://leetcode.com/problems/find-right-interval/discuss/91806/Python-O(nlogn)-short-solution-with-explanation
"""
class Solution:
    def findRightInterval(self, intervals: List[List[int]]) -> List[int]:
        n = len(intervals)
        starting_pt = sorted([(intervals[i][0], i) for i in range(n)]);
        res = [];
        for interval in intervals:
            r = bisect.bisect_left(starting_pt, (interval[1], ))
            res.append(starting_pt[r][1] if r != n else -1)
        return res

File: 436.leetcode.cpp
/**
436. Find Right Interval
https://leetcode.com/problems/find-right-interval/
**/
class Solution {
public:
    vector<int> findRightInterval(vector<vector<int>>& intervals) {
        vector<pair<vector<int>, int>> interval_list;
        int n = intervals.size();
        vector<int> res(n);
        for (int i = 0; i < n; ++i) {
            interval_list.push_back(make_pair(intervals[i], i));
        }
        sort(interval_list.begin(), interval_list.end());

        for (int j = 0; j < n; ++j) {
            int min_index = n;
            for (int k = j + 1; k < n; ++k) {
                if (interval_list[k].first[0] >= interval_list[j].first[1]) {
                    min_index = interval_list[k].second;
                    break;
                }
            }
            res[interval_list[j].second] = min_index == n ? -1 : min_index;
        }
        return res;

    }
};
